tests/tr1/tests/cwchar1/test.cpp

```diff
@@ -141,19 +140,15 @@ void test_cpp() { // test C++ header
 
         CHECK(wmacs[1] < wmacs[0]);
 
-#if NO_TMPNAM_TESTS
-        char *tname, *tn;
-        assert((tn = CSTD tmpnam((char*) nullptr)) != nullptr);
+        char* tname;
+        const char* tn;
+        const auto temp_name1 = temp_file_name();
+        assert((tn = temp_name1.c_str()) != nullptr);
         tname = (char*) CSTD malloc(CSTD strlen(tn) + 1);
         CSTD strcpy(tname, tn);
 
-#else // NO_TMPNAM_TESTS
-        char tname[L_tmpnam], *tn;
-        CHECK_PTR(CSTD tmpnam(tname), tname);
-        assert(CSTD strlen(tname) < L_tmpnam);
-#endif // NO_TMPNAM_TESTS
-
-        assert((tn = CSTD tmpnam((char*) nullptr)) != nullptr);
+        const auto temp_name2 = temp_file_name();
+        assert((tn = temp_name2.c_str()) != nullptr);

```

Comment:
Ditto "three occurrences of `std::string::c_str() != nullptr`". And one occurrence in each of the following two tests. (I won't block the review for these, but let's just say I wouldn't mind reviewing more changes here =))

Reply:
Fixed these occurrences and the following two tests.